python/psautohint/__init__.py

```diff
@@ -31,18 +30,19 @@ class FontParseError(Exception):
 
 
 def _font_is_ufo(path):
-    if os.path.isdir(path) and path.lower().endswith('.ufo'):
-        meta_path = os.path.join(path, 'metainfo.plist')
-        if os.path.isfile(meta_path):
-            metainfo = plistlib.readPlist(meta_path)
-            keys = ('creator', 'formatVersion')
-            if metainfo and all([key in metainfo for key in keys]):
-                return True
-    return False
+    from fontTools.ufoLib import UFOReader
+    from fontTools.ufoLib.errors import UFOLibError
+    try:
+        UFOReader(path, validate=False)
+        return True
+    except (UFOLibError, KeyError, TypeError):
```

Comment:
@anthrotype Having to catch `KeyError` and `TypeError` here is a bit ugly I think, not sure if I should have catch `Exception` instead. `UFOReader.readMetaInfo()` tries to access items in the parsed dict unconditionally, but it can be None (empty `metainfo.plist`) or misses `formatVersion`, I’d have expected `UFOLibError` in these cases.

Reply:
No it doesn’t: https://github.com/fonttools/fonttools/blob/76c85e3c3b6f2fa0fa4a8bdec7036b4bb06c6154/Lib/fontTools/ufoLib/__init__.py#L392-L395

Reply:
Ok, we need to fix that.
An except Exception: is not that bad in this particular case, but this also works